### PR TITLE
Sync Pause and Resume, enable analyzers

### PR DESCRIPTION
**Pull request by Shadowghost**
* Properly post play/pause to trakt
* Secure API
* Enable all analyzers and fix all warnings

### CODE REVIEW

File: Trakt/Api/DataContracts/BaseModel/TraktEpisode.cs [hunk 1]
```diff
@@ -1,30 +1,34 @@
 using System.Text.Json.Serialization;
 
-namespace Trakt.Api.DataContracts.BaseModel;
```
Comment (crobibero) [inline Trakt/Api/DataContracts/BaseModel/TraktEpisode.cs:3]: Why move away from file-scoped space?
That's the newer way to do things

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces

I can re-clean it up in a future PR, Rider makes it very easy to do for the entire solution

File: Trakt/Model/TraktAudio.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+{
+    /// <summary>
+    /// Enum TraktAudio.
+    /// </summary>
+    public enum TraktAudio
+    {
+        /// <summary>
+        /// LPCM audio.
+        /// </summary>
+        [Description("lpcm")]
```
Comment (crobibero) [inline Trakt/Model/TraktAudio.cs:13]: I don't believe the `Description` overrides what is returned in the `.ToString()` call

File: Trakt/Api/DataContracts/BaseModel/TraktEpisode.cs (hunk 1 above)
Comment (Shadowghost) [reply Trakt/Api/DataContracts/BaseModel/TraktEpisode.cs:3]: Sounds good. I just adapted the style from the main Jellyfin codebase...
I was already nearly done when I thought about looking up which style is current best practices :/

File: Trakt/Model/TraktAudio.cs (hunk 2 above)
Comment (Shadowghost) [reply Trakt/Model/TraktAudio.cs:13]: You're correct I misread the stackoverflow. I'll switch that back and supress the warning.

File: Trakt/Api/TraktController.cs [hunk 3]
```diff
@@ -6,152 +6,160 @@
...
+        /// <summary>
+        /// Rate an item.
+        /// </summary>
+        /// <param name="userId">The user id.</param>
+        /// <param name="itemId">The item id.</param>
+        /// <param name="rating">Rating between 1 - 10 (0 = unrate).</param>
+        /// <response code="200">Item rated successfully.</response>
+        /// <returns>A <see cref="TraktSyncResponse"/>.</returns>
+        [HttpPost("Users/{userId}/Items/{itemId}/Rate")]
+        [Authorize(Policy = "DefaultAuthorization")]
```
Comment (crobibero) [inline Trakt/Api/TraktController.cs:120]: This isn't needed as it's already being applied at the controller level
`RequiresElevation` should stay though

**Review by crobibero — APPROVED** [review]

**Pull request merged**